core/src/com/unciv/ui/pickerscreens/FoundReligionPickerScreen.kt

```diff
@@ -89,11 +92,84 @@ class FoundReligionPickerScreen (
                 rightSideButton.label = "Found [$translatedReligionName]".toLabel()
                 checkAndEnableRightSideButton()
             }
-            if (religionName == this.religionName) button.disable()
+            if (religionName == this.religionName || gameInfo.religions.keys.any { it == religionName }) button.disable()
             iconsTable.add(button).pad(5f)
         }
         iconsTable.row()
         topReligionIcons.add(iconsTable).padBottom(10f).row()
-        topReligionIcons.add(descriptionLabel).center()
+        topReligionIcons.add(descriptionLabel).center().padBottom(5f)
+    }
+
+    private fun updateLeftTable() {
+        leftChosenBeliefs.clear()
+        val currentReligion = choosingCiv.religionManager.religion ?: Religion("Unknown", gameInfo, choosingCiv.civName)
+        
+        for (pantheon in currentReligion.getPantheonBeliefs()) {
+            val beliefButton = convertBeliefToButton(pantheon)
+            leftChosenBeliefs.add(beliefButton).pad(10f).row()
+            beliefButton.disable()
```

Comment:
Looks like youre doing the same thing twice. So maybe just for (belief in currentReligion.getFollowerBeliefs() + currentReligion.getPantheonBeliefs())